upgrade.go

```diff
@@ -43,6 +38,32 @@ func (b *versionedKVBackend) Upgrade(ctx context.Context, s logical.Storage) err
 		return errors.New("upgrade already in process")
 	}
 
+	// If we are a replication secondary, wait until the primary has finished
+	// upgrading.
+	if !b.System().LocalMount() && b.System().ReplicationState().HasState(consts.ReplicationPerformanceSecondary) {
+		b.Logger().Info("upgrade not running on performace replication secondary")
```

Comment:
Just to make sure I'm following, upgrade check happens on plugin creation which will be redone when the mount table changes due to invalidation. So it will go into this upgrade loop, then wait for it to be done, then set the uint back.

Reply:
Yep, that's the idea. I think it will require a slight change to the invalidation logic but will test it out once this is merged.